### PR TITLE
Syncer abort all tasks from a certain source

### PR DESCRIPTION
**Pull request by TheCharlatan**

### CODE REVIEW

**Review by zkao — CHANGES_REQUESTED** [review]
Small suggestions on comments

File: src/syncerd/syncer_state.rs [hunk 1]
```diff
@@ -16,18 +16,38 @@ use crate::serde::Deserialize;
 use crate::syncerd::*;
 use hex;
 
+#[derive(Debug, Clone, Copy, Eq, PartialEq, PartialOrd, Hash, Display)]
+#[display(Debug)]
+pub struct InternalId(u32);
```
[the remark this answers is not in the document]
Comment (TheCharlatan) [reply src/syncerd/syncer_state.rs:21]: The idea is that you can always increment the `TaskCounter`, but you can never accidentally change the `InternalId`.

File: src/syncerd/syncer_state.rs [hunk 2]
```diff
@@ -66,152 +86,149 @@ impl SyncerState {
...
+            task_count: TaskCounter(0),
         }
     }
 
     pub fn block_height(&self) -> u64 {
         self.block_height
     }
 
-    pub async fn abort(&mut self, task_id: u32, source: ServiceId) {
+    pub async fn abort(&mut self, task_id: Option<u32>, source: ServiceId) {
```
[the remark this answers is not in the document]
Comment (TheCharlatan) [reply src/syncerd/syncer_state.rs:97]: Yeah, fine by me. Will change.

[the remark this answers is not in the document]
Comment (TheCharlatan) [reply src/syncerd/syncer_state.rs]: Done in https://github.com/farcaster-project/farcaster-node/pull/198/commits/0cb1c3fe051b38a0d556333a7b4d92bbfa2eebb8

File: src/syncerd/types.rs [hunk 3]
```diff
@@ -60,7 +60,13 @@ pub struct SweepBitcoinAddress {
 #[derive(Clone, Debug, Display, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
 #[display(Debug)]
 pub struct Abort {
-    pub id: u32,
+    pub task_id_or_all_tasks: TaskIdOrAllTasks,
+}
+
+#[derive(Clone, Debug, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
+pub enum TaskIdOrAllTasks {
```
Comment (h4sh3d) [inline src/syncerd/types.rs:67]: Can we rename it to something like `TaskTarget` ;p ?

**Review by h4sh3d — APPROVED** [review]

**Review by zkao — APPROVED** [review]

File: src/syncerd/types.rs (hunk 3 above)
Comment (TheCharlatan) [reply src/syncerd/types.rs]: Done in 49bd62ad92535b759936536880f457e13ead24ba

**Review by h4sh3d — APPROVED** [review]

**Pull request merged**